sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -189,17 +190,35 @@ trait V2CreateTablePlan extends LogicalPlan {
   def withPartitioning(rewritten: Seq[Transform]): V2CreateTablePlan
 }
 
+trait V2CreateTablePlanX extends LogicalPlan {
```

Comment:
`V2CreateTablePlan` can't be deleted yet because it is also extended by `CreateTableAsSelect`, `ReplaceTable` and `ReplaceTableAsSelect`. I use `V2CreateTablePlanX` for this new trait for now.

Reply:
Do we have to add this? Can we let the new `CreateTable` extends `V2CreateTablePlan` and implement `tableName`?
```
def tableName = {
  assert(name.isResolved)
  name.asInstanceOf[ResolvedDBObjectName].nameParts.asIdentifier
}
```